plugins/base/src/main/kotlin/signatures/KotlinSignatureProvider.kt

```diff
@@ -64,7 +63,12 @@ class KotlinSignatureProvider(ctcc: CommentsToContentConverter, logger: DokkaLog
         list(f.parameters) {
             link(it.name!!, it.dri)
             text(": ")
-            type(it.type)
+
+            val type = it.type
+            when {
+                type is KotlinTypeWrapper && type.isFunctionType -> funType(type)
```

Comment:
A riddle?  What is a 'when' with two options only?
Answer: An 'if'

Reply:
Changed.